File: Java/binarysearch.java
// Binary search
// prerequisite-Sorted arrays

public class binarysearch {
    public static int func(int number[], int key) {

        int start = 0;
        int end = number.length - 1;
        while (start <= end) {
            int mid = (start + end) / 2;
            if (number[mid] == key) {
                return mid;
            }
            if (number[mid] < key) {
                start = mid + 1;
            } else {
                end = mid - 1;
            }
        }
        return -1;
    }

    public static void main(String args[]) {
        int number[] = { 45, 8, 34, 97, 23, 54, 23, 67 };
        int key = 345;

        System.out.print("Index of given key is :" + func(number, key));
    }
}

File: c++ DSA/BinarySearch.cpp
#include <bits/stdc++.h>

using namespace std;

int binarySearch(int arr[] , int size , int key){
    int start = 0;
    int end = size-1;

    int mid = start + (end- start)/2;

    while(start<=end){
        if(arr[mid]==key){
            return mid;
        }

        if(key>arr[mid]){
            start = mid+1;
        }
        else{
            end = mid -1;
        }
        
        mid = start + (end- start)/2;
    }
    return -1;
}

int main(){
    int even[6] = {3,5,6,8,10,15};
    int odd[5] = {4,6,12, 23,30};


    int index = binarySearch(even , 6 , 10);
    cout << " the index of 10 is  "<< index << endl;
}